slsa_for_models/README.md

```diff
@@ -0,0 +1,85 @@
+# SLSA for Models
+
+To protect the supply chain of traditional software against tampering (like in
+the [Solarwinds attack][solarwinds]), we can generate SLSA provenance, for
+example by using the [SLSA L3 GitHub generator][slsa-generator].
+
+This project shows how we can use the [SLSA L3 GitHub generator][slsa-generator]
+to generate SLSA provenance for ML models. The SLSA generator was originally
+developed for traditional software to protect against tampering with builds,
+such as in the [Solarwinds attack][solarwinds], and this project is a proof of
+concept that the _same supply chain protections can be applied to ML_.
+
+Future work will involve covering training ML models that require access to
+accelerators (i.e., GPUs, TPUs) or that require multiple hours for training. As
+an example, we could use [Tekton Chains][tekton-chains] to support [training ML
+models using Kubeflow][tekton-kubeflow].
+
+When users download a given version of a model they can also check its
+provenance by using [the SLSA verifier][slsa-verifier] repository. This can be
+done automatically: for example the model serving pipeline could validate
+provenance for all new models before serving them. The verification can also be
+done manually, on demand.
+
+As an additional benefit, having provenance for a model allows users to react
+to vulnerabilities in a training framework: they can quickly determine if a
+model needs to be retrained because it was created using a vulnerable version.
+
+## Usage
+
+We support both TensorFlow and PyTorch models. The example repo trains a model
```

Comment:
do we add a disclaimer here about file vs directory models?

Reply:
I added a note at the end